src/languages/html.coffee

```diff
@@ -0,0 +1,684 @@
+
+    #console.log node
+
+    node.__indentLocation = {start: node.__location.startTag.end}
+
+    if node.childNodes?.length > 0
+      node.__indentLocation.end = node.childNodes[node.childNodes.length - 1].__location.end
+    else
+      node.__indentLocation.end = node.__location.startTag.end
+
+    if not node.__location.endTag
+      node.__location.end = node.__indentLocation.end
+
+    @setAttribs node, @text[node.__location.start...node.__indentLocation.start]
+
+  makeIndentBounds: (node) ->
+    ###
+    loc = node.__indentLocation
+    lines = @text[loc.start...loc.end].split('\n')
+    for i in [lines.length-1..0] by -1
+      if lines[i].trim().length is 0
+        loc.end -= lines[i].length + 1
+      else
+        break
+
+    if loc.start > loc.end
+      loc.end = loc.start
+    ###
+    bounds = {
+      start: @positions[node.__indentLocation.start]
+      end: @positions[node.__indentLocation.end]
+    }
+
+    if node.__location.endTag?
+      lastLine = @positions[node.__location.endTag.start].line - 1
+      if lastLine > bounds.end.line or (lastLine is bounds.end.line and @lines[lastLine].length > bounds.end.column)
+        bounds.end = {
+          line: lastLine
+          column: @lines[lastLine].length
+        }
+      else if node.__indentLocation.start is node.__indentLocation.end and node.__location.endTag
+        bounds.end = @positions[node.__location.endTag.start]
+
+    return bounds
+
+  trimText: (node) ->
+    text = node.value
+    location = node.__location
+    text = text.split '\n'
+    i = 0
+    while text[i].trim().length is 0
+      location.start += text[i].length + 1
+      i++
+    j = text.length - 1
+    while text[j].trim().length is 0
+      j--
+    text = text[i..j].join '\n'
+    location.end = location.start + text.length
+    if i isnt 0
+      leftTrimText = text.trimLeft()
+      location.start += text.length - leftTrimText.length
+      text = leftTrimText
+    node.value = text
+    ###
+    rigthtTrimText = text.trimRight()
+    leftTrimText = rigthtTrimText.trimLeft()
+    node.value = leftTrimText
+    location.start += rigthtTrimText.length - leftTrimText.length
+    location.end = location.start + leftTrimText.length
+    console.log node.value, JSON.stringify location
+    ###
+
+  getSocketLevel: (node) -> helper.ANY_DROP
+
+  htmlBlock: (node, depth, bounds) ->
+    #console.log "Adding Block: ", JSON.stringify(bounds ? @getBounds node)
+    @addBlock
+      bounds: bounds ? @getBounds node
+      depth: depth
+      precedence: @getPrecedence node
+      color: @getColor node
+      classes: @getClasses node
+      socketLevel: @getSocketLevel node
+
+  htmlSocket: (node, depth, precedence, bounds, classes) ->
+    #console.log "Adding Socket: ", JSON.stringify(bounds ? @getBounds node)
+    @addSocket
+      bounds: bounds ? @getBounds node
+      depth: depth
+      precedence: precedence
+      classes: classes ? @getClasses node
+      acccepts: @getAcceptsRule node
```

Comment:
"acccepts" is probably a typo, although the "accepts" argument to "addSocket" is actually no longer required (we should probably update the README).